From: Vito Caputo <vcaputo@pengaru.com>
Date: Sat, 3 Feb 2018 03:09:08 -0800
Subject: socket-util: fix getpeergroups() assert(fd) (#8080)

Don't assert on zero-value fds.

Fixes #8075.

(cherry picked from commit 75f40779607ea79f20441c7fb46744d04ee2c7ae)
---
 src/basic/socket-util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/basic/socket-util.c b/src/basic/socket-util.c
index 2c70cad..bd3cf6f 100644
--- a/src/basic/socket-util.c
+++ b/src/basic/socket-util.c
@@ -1008,7 +1008,7 @@ int getpeergroups(int fd, gid_t **ret) {
         socklen_t n = sizeof(gid_t) * 64;
         _cleanup_free_ gid_t *d = NULL;
 
-        assert(fd);
+        assert(fd >= 0);
         assert(ret);
 
         for (;;) {
